peer/addrinfo.go

```diff
@@ -29,11 +29,7 @@ func AddrInfosFromP2pAddrs(maddrs ...ma.Multiaddr) ([]AddrInfo, error) {
 		if id == "" {
 			return nil, ErrInvalidAddr
 		}
-		if transport == nil {
-			if _, ok := m[id]; !ok {
-				m[id] = nil
-			}
```

Comment:
This was necessary. Just below this, we iterate over `m` so `id` needs to be a key in the map.

Reply:
Would you consider adding a test for this?

Reply:
Sorry, you are right, when I changed the code and run the testcase: TestAddrInfosFromP2pAddrs, it passed. But the case does not show this problem and I do find this PR is wrong, I'll close it.